### PR TITLE
Make Job.location optional

### PR DESCRIPTION
**Pull request by wslulciuc**
This PR makes the `location` of a given job _optional_.

### CODE REVIEW

File: src/main/java/marquez/service/models/JobMeta.java [hunk 1]
```diff
@@ -74,7 +78,7 @@ public UUID version(@NonNull NamespaceName namespaceName, @NonNull JobName jobNa
                     .map(output -> output.getValue())
                     .collect(joining(VERSION_DELIM)),
                 KV_JOINER.join(getContext()),
-                location.toString())
+                getLocation().map(location -> location.toString()).orElse(null))
```
Comment (roaraya8) [inline src/main/java/marquez/service/models/JobMeta.java]: you could do `URL::toString` here

File: src/main/java/marquez/api/mappers/Mapper.java [hunk 2]
```diff
@@ -184,7 +184,7 @@ public static JobResponse toJobResponse(@NonNull final Job job) {
         ISO_INSTANT.format(job.getUpdatedAt()),
         job.getInputs().stream().map(input -> input.getValue()).collect(toImmutableList()),
         job.getOutputs().stream().map(output -> output.getValue()).collect(toImmutableList()),
-        job.getLocation().toString(),
+        job.getLocation().map(location -> location.toString()).orElse(null),
```
Comment (roaraya8) [inline src/main/java/marquez/api/mappers/Mapper.java]: you could use `.map(URL::toString)` here as well

File: src/main/java/marquez/service/models/JobMeta.java (hunk 1 above)
Comment (wslulciuc) [reply src/main/java/marquez/service/models/JobMeta.java]: Yes! And hand no idea this was an option 💯

File: src/main/java/marquez/api/mappers/Mapper.java (hunk 2 above)
Comment (wslulciuc) [reply src/main/java/marquez/api/mappers/Mapper.java]: Done!

**Pull request merged**